routes.go

```diff
@@ -0,0 +1,27 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// alphanum20 is a regular expression matching an sequence of
+// 20 alphanumeric characters.
+const alphanum20 = "[a-zA-Z0-9]{2,20}"
+
+func (s *Server) registerRoutes() {
+	s.router.HandleFunc("/", handleRoot)
+
+	v1 := s.router.PathPrefix("/v1").Subrouter()
+
+	v1.HandleFunc("/report", s.createReport).Methods("POST")
+
+	v1.HandleFunc("/report", s.retrieveReport).Methods("GET").
+		Queries("id", fmt.Sprintf("{id:%s}", alphanum20))
```

Comment:
I'm pretty sure it results in 405 Method Not Allowed ! At least that's what I got in may local tests.

`/report/{id}` is fine too but it would be `reports/{id}` plural. And we would maybe reflect that to the POST endpoint. I see no objection, but no requirement either.

Reply:
Hah I should have checked before! Well it's even worse then, 405 is clearly incorrect.
`reports/{id}` seems more conventional to me.